### DISCUSSION TITLE
Modularisation and views

### DISCUSSION THREAD

**Discussion by arthurcro** [pointfreeco/isowords#173] [Q&A]
Hey there! :wave: 

First, thank you for all the open source work you have been doing at pointfree.co. It's been a great source of inspiration and an amazing learning resource for me the past couple years! 

I'm working on a large modularised codebase mostly built in Vanilla SwiftUI. I have several modules that encapsulate some features that are reused in multiple (If not all in some cases) feature modules.

I'm currently importing those modules directly into my feature modules when I need them but that causes an increased build time and transitively import dependencies that are really only implementation details of those reusable feature modules. A similar problem was brought up during the [SwiftUI Navigation: Decoupling](https://www.pointfree.co/collections/swiftui/navigation/ep212-swiftui-navigation-decoupling#t113) episode I believe.

I was wondering if you explored any ideas to reduce this coupling? I thought about a similar approach used for dependencies, separating interface and implementation in different modules, but I'm struggling writing a good interface for this.  

Thank you! 
Arthur

**Comment by beetee17** [comment]
Hi @arthurcro, was wondering if you have made any progress since the original post? I've also encountered similar questions recently. 

I made [a post on the TCA repo's discussion page](https://github.com/pointfreeco/swift-composable-architecture/discussions/3143#discussioncomment-9680542) if you're interested in checking out the solution I've been playing around with, though I presume you've grappled with this problem far longer than I have, so I'd love to hear your thoughts!
